crates/op-rbuilder/src/generator.rs

```diff
@@ -68,6 +69,10 @@ pub struct BlockPayloadJobGenerator<Client, Pool, Tasks, Builder> {
     ///
     /// See [PayloadBuilder]
     builder: Builder,
+    /// Whether to ensure only one payload is being processed at a time
+    ensure_only_one_payload: bool,
+    /// The last payload being processed
+    last_payload: Arc<Mutex<Option<CancellationToken>>>,
```

Comment:
nitpick: since you always cancel you can just have value here all the time without any option